src/main/java/com/openshift/jenkins/plugins/pipeline/model/IOpenShiftDeployer.java

```diff
@@ -48,7 +48,8 @@ default boolean coreLogic(Launcher launcher, TaskListener listener, Map<String,
             IReplicationController rc = null;
             long wait = getTimeout(listener, chatty, overrides);
             while (System.currentTimeMillis() < (currTime + wait)) {
-                dc = client.get(ResourceKind.DEPLOYMENT_CONFIG, getDepCfg(overrides), getNamespace(overrides));
+                if (dc == null)
+                    dc = client.get(ResourceKind.DEPLOYMENT_CONFIG, getDepCfg(overrides), getNamespace(overrides));
                 if (dc != null) {
                     if (!versionBumped) {
```

Comment:
The timing window manifested in two spots.

1) First, the manual version bump update did not always take hold *immediately*, so versionBumped stayed false.  Here is a snippet of the debug I added:

```
OpenShiftDeployer latest version now 1 and was 1
```

This always coincided with not being able to get the RC:

```
com.openshift.restclient.NotFoundException: replicationcontrollers "myphp-1" not found replicationcontrollers "myphp-1" not found
	at com.openshift.internal.restclient.okhttp.ResponseCodeInterceptor.createOpenShiftException(ResponseCodeInterceptor.java:112)
	at com.openshift.internal.restclient.okhttp.ResponseCodeInterceptor.intercept(ResponseCodeInterceptor.java:65)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:170)
	at okhttp3.RealCall.execute(RealCall.java:60)
	at com.openshift.internal.restclient.DefaultClient.execute(DefaultClient.java:217)
	at com.openshift.internal.restclient.DefaultClient.execute(DefaultClient.java:194)
	at com.openshift.internal.restclient.DefaultClient.execute(DefaultClient.java:183)
	at com.openshift.internal.restclient.DefaultClient.get(DefaultClient.java:291)
	at com.openshift.jenkins.plugins.pipeline.model.IOpenShiftPlugin.getLatestReplicationController(IOpenShiftPlugin.java:565)
	at com.openshift.jenkins.plugins.pipeline.model.IOpenShiftDeployer.coreLogic(IOpenShiftDeployer.java:79)
```

Were it got tricky was that after sleeping for a few seconds, the DC bump did occur, and when we repopulated `dc`, it had in fact now had the update (now version 2).  The next call to bump still had the delay, and now versionBumped still was kept at false because now 2 was not greater than 2.

The code change essentially ensures that the "original" dc is kept in the `dc` variable, and whenever we get a dc with a greater version, we can proceed.